qt/src/main.rs

```diff
@@ -95,3 +132,28 @@ fn main() {
         },
     }
 }
+
+fn filepath_from_args(args_vec: Vec<String>) -> Option<String> {
+    match args_vec.into_iter().next(){
+        Some(filepath) => {
+            if filepath == "--help" || filepath == "-h"{
+                println!("{}", USAGE);
+                return None;
+            }
+            match File::open(&filepath){
+                Ok(_) => {
+
+                    return Some(filepath);
+                },
+                Err(e) => {
+                    println!("IO Error: {}", e);
+                    return None;
+                }
+            }
+        },
+        None => {
+            println!("{}", USAGE);
+            return None;
+        }
+    }
+}
```

Comment:
Also we can always take a `Path` object to determine wether the input is a file, a directory or a green elephant 🐘 🤣